src/backend/distributed/commands/extension.c

```diff
@@ -515,19 +515,70 @@ MarkExistingObjectDependenciesDistributedIfSupported()
 
 		if (ShouldSyncTableMetadata(citusTableId))
 		{
-			/* we need to pass pointer allocated in the heap */
-			ObjectAddress *addressPointer = palloc0(sizeof(ObjectAddress));
-			*addressPointer = tableAddress;
+			/*
```

Comment:
Probably a separate thing to look. Considering https://github.com/citusdata/citus/pull/4344, maybe we should use `ShouldSyncTableMetadataViaCatalog` instead of `ShouldSyncTableMetadata` ?

Worth testing an upgrade with 100k distributed tables.

Reply:
Not super critical impact, but still I'd be in favor of merging this https://github.com/citusdata/citus/pull/5925

Reply:
already merged